main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala

```diff
@@ -0,0 +1,489 @@
+
+    object Expr {
+
+      /**
+       * A marker kind used to wrap nested expressions.
+       * For instance on a binary expression "1 + 2" you would do
+       * Expr
+       * Binary
+       * Expr
+       * LiteralNumber
+       * Operator
+       * Expr
+       * LiteralNumber
+       */
+      case object Expr extends Expr
+
+      case object Apply extends Expr
+
+      case object Ascribe extends Expr
+
+      case object Binary extends Expr
+
+      case object Block extends Expr
+
+      case object CheckedEffectCast extends Expr
+
+      case object CheckedTypeCast extends Expr
+
+      case object Do extends Expr
+
+      case object FixpointConstraint extends Expr
+
+      case object FixpointConstraintSet extends Expr
+
+      case object FixpointFromFragment extends Expr
+
+      case object FixpointInject extends Expr
+
+      case object FixpointQuery extends Expr
+
+      case object FixpointSelect extends Expr
+
+      case object FixpointSolve extends Expr
+
+      case object FixpointWhere extends Expr
+
+      case object ForApplicative extends Expr
+
+      case object Foreach extends Expr
+
+      case object ForeachYield extends Expr
+
+      case object ForMonadic extends Expr
+
+      case object Generator extends Expr
+
+      case object Guard extends Expr
+
+      case object Hole extends Expr
+
+      case object HoleVariable extends Expr
+
+      case object IfThenElse extends Expr
+
+      case object InstanceOf extends Expr
+
+      case object Intrinsic extends Expr
+
+      case object JvmMethod extends Expr
+
+      case object Lambda extends Expr
+
+      case object LambdaMatch extends Expr
+
+      case object LetImport extends Expr
+
+      case object LetMatch extends Expr
+
+      case object LetRecDef extends Expr
+
+      case object Literal extends Expr
+
+      case object LiteralArray extends Expr
+
+      case object LiteralList extends Expr
+
+      case object LiteralMap extends Expr
+
+      case object LiteralMapKeyValueFragment extends Expr
+
+      case object LiteralRecord extends Expr
+
+      case object LiteralRecordFieldFragment extends Expr
+
+      case object LiteralSet extends Expr
+
+      case object LiteralVector extends Expr
+
+      case object Match extends Expr
+
+      case object MatchRuleFragment extends Expr
+
+      case object NewObject extends Expr
+
+      case object OpenVariant extends Expr
+
+      case object OpenVariantAs extends Expr
+
+      case object Paren extends Expr
+
+      case object ParYield extends Expr
+
+      case object ParYieldFragment extends Expr
+
+      case object RecordOperation extends Expr
+
+      case object RecordOpExtend extends Expr
+
+      case object RecordOpRestrict extends Expr
+
+      case object RecordOpUpdate extends Expr
+
+      case object RecordSelect extends Expr
+
+      case object Ref extends Expr
+
+      case object Scope extends Expr
+
+      case object ScopeName extends Expr
```

Comment:
What is this?

Reply:
That refers to the "scope" or region part following some expressions
```
ref e @ rc
        ^^ This part here
```
I use it to mark what expression is the scope name (so the weeded doesn't confuse  `e` with `rc`)